src/operators_lazysum.jl

```diff
@@ -104,6 +104,44 @@ end
 -(a::O1, b::O2) where {O1<:LazyOperator,O2<:LazyOperator} = LazySum(a) - LazySum(b)
 
 
+#*(a::LazySum{B1,B2}, b::AbstractOperator{B2,B3}) where {B1,B2,B3} = LazySum(a.basis_l, a.basis_r, a.factors, a.operators .* b)
+#*(a::AbstractOperator{B1,B2}, b::LazySum{B2,B3}) where {B1,B2,B3} = LazySum(a.basis_l, b.basis_r, b.factors, a.operators .* b)
+
+#*(a::Operator{B1,B2}, b::LazySum{B2,B3}) where {B1,B2,B3} = LazySum(a.basis_l, b.basis_r, b.factors, a.operators .* b)
+#*(a::LazySum{B1,B2}, b::Operator{B2,B3}) where {B1,B2,B3} = LazySum(a.basis_l, b.basis_r, a.factors, a.operators .* b)
+
+
+function Base.:*(a::LazySum{B1,B2}, b::O2) where {B1,B2,O2<:LazyOperator}
+    c = Array{AbstractOperator}(undef,length(a.operators))
+    for i in eachindex(a.operators)
+        c[i] = LazyProduct(a.operators[i]) * b
+    end
+    LazySum(a.basis_l, b.basis_r, a.factors, (c...,))
```

Comment:
You have an array typed to `AbstractOperator`. If your array is already meant to contain abstract types, there is not much reason to do anything more specific than `Any`, especially given that it would be slightly more readable code.

More importantly though, here you create a temporary array and then you immediately destroy it. It is a bit wasteful (causing unnecessary allocations). It might be a bit simpler to just use a "generator" (a "lazy" object that does not actually compute the content until it is queried):

```
julia> generator = (i for i in 1:10)
Base.Generator{UnitRange{Int64}, typeof(identity)}(identity, 1:10)

julia> tuple = (generator...,)
(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
``` 

So your code can be simplified to something like `LazySum(bl, br, a.factors, ((LazyProduct(op)*b for op in a.operators)...,))`